src/py/run_generator.rs

```diff
@@ -0,0 +1,290 @@
+        );
+    }
+
+    if spend_cond.height_relative > 0 {
+        ordered.insert(
+            ASSERT_HEIGHT_RELATIVE,
+            make_condition(ASSERT_HEIGHT_RELATIVE, spend_cond.height_relative as u64),
+        );
+    }
+
+    if spend_cond.height_absolute > 0 {
+        ordered.insert(
+            ASSERT_HEIGHT_ABSOLUTE,
+            make_condition(ASSERT_HEIGHT_ABSOLUTE, spend_cond.height_absolute as u64),
+        );
+    }
+
+    if spend_cond.seconds_relative > 0 {
+        ordered.insert(
+            ASSERT_SECONDS_RELATIVE,
+            make_condition(ASSERT_SECONDS_RELATIVE, spend_cond.seconds_relative as u64),
+        );
+    }
+
+    if spend_cond.seconds_absolute > 0 {
+        ordered.insert(
+            ASSERT_SECONDS_ABSOLUTE,
+            make_condition(ASSERT_SECONDS_ABSOLUTE, spend_cond.seconds_absolute as u64),
+        );
+    }
+
+    let mut conditions = Vec::<(ConditionOpcode, Vec<PyConditionWithArgs>)>::new();
+    for (k, v) in ordered {
+        conditions.push((k, v));
+    }
+
+    PyNPC {
+        coin_name: PyBytesObj {
+            arr: spend_cond.coin_id.to_vec(),
+        },
+        puzzle_hash: PyBytesObj {
+            arr: Vec::<u8>::from(a.atom(spend_cond.puzzle_hash)),
+        },
+        conditions,
+    }
+}
+
+impl IntoPy<PyObject> for ErrorCode {
+    fn into_py(self, py: Python) -> PyObject {
+        let ret = match self {
+            ErrorCode::NegativeAmount => 124,
+            ErrorCode::InvalidPuzzleHash => 10,
+            ErrorCode::InvalidPubkey => 10,
+            ErrorCode::InvalidMessage => 10,
+            ErrorCode::InvalidParentId => 10,
+            ErrorCode::InvalidConditionOpcode => 10,
+            ErrorCode::InvalidCoinAnnouncement => 10,
+            ErrorCode::InvalidPuzzleAnnouncement => 10,
+            ErrorCode::InvalidCondition => 10,
+            ErrorCode::InvalidCoinAmount => 10,
+            ErrorCode::AssertHeightAbsolute => 14,
+            ErrorCode::AssertHeightRelative => 13,
+            ErrorCode::AssertSecondsAbsolute => 15,
+            ErrorCode::AssertSecondsRelative => 105,
+            ErrorCode::AssertMyAmountFailed => 116,
+            ErrorCode::AssertMyPuzzlehashFailed => 115,
+            ErrorCode::AssertMyParentIdFailed => 114,
+            ErrorCode::AssertMyCoinIdFailed => 11,
+            ErrorCode::AssertPuzzleAnnouncementFailed => 12,
+            ErrorCode::AssertCoinAnnouncementFailed => 12,
+            ErrorCode::ReserveFeeConditionFailed => 48,
+            ErrorCode::DuplicateOutput => 4,
+            ErrorCode::DoubleSpend => 5,
+            ErrorCode::CostExceeded => 23,
+        };
+        ret.to_object(py)
+    }
+}
+
+// returns the cost of running the CLVM program along with the list of NPCs for
+// the generator/spend bundle. Each SpendConditionSummary is a coin spend along with its
+// conditions
+#[allow(clippy::too_many_arguments)]
+#[pyfunction]
+pub fn run_generator(
```

Comment:
Could we simply call `run_program` from python and then feed the `LazyNode` into a simplified version of this function starting at line 251? The upside is we would have less redundant api. The downside is the python could in theory retain a careless reference to the lazy node, delaying destroying the allocator (but as the doctor says when you tell him "it hurts when I do this", "Well then, don't do that.").

Reply:
I think it's a net increase in complexity to do that. Especially to ensure we don't accidentally do something expensive, either CPU- or memory wise.